pkg/kubeapiserver/options/authentication.go

```diff
@@ -281,7 +299,17 @@ func (s *BuiltInAuthenticationOptions) AddFlags(fs *pflag.FlagSet) {
 
 		fs.StringVar(&s.ServiceAccounts.Issuer, "service-account-issuer", s.ServiceAccounts.Issuer, ""+
 			"Identifier of the service account token issuer. The issuer will assert this identifier "+
-			"in \"iss\" claim of issued tokens. This value is a string or URI.")
+			"in \"iss\" claim of issued tokens. This value is a string or URI. If this option is not "+
+			"a valid URI per the OpenID Discovery 1.0 spec, the ServiceAccountIssuerDiscovery feature "+
+			"will remain disabled, even if the feature gate is set to true. It is highly recommended "+
+			"that this value comply with the OpenID spec: https://openid.net/specs/openid-connect-discovery-1_0.html.")
```

Comment:
most people reading this won't know what that means... would it make sense to describe the explicit requirements we care about, like:
* "This must be a https URL in order for discovery information to be served at `/.well-known/openid-configuration`"
* "It is highly recommended this be set to a URL that is capable of serving openid discovery documents at `${service-account-issuer}/.well-known/openid-configuration`"